File: schemas/crm/drizzle/leads.ts
// leads: inbound prospects before conversion to contacts.
// See README.md for full design rationale.

import { pgTable, pgEnum, uuid, text, timestamp, index } from "drizzle-orm/pg-core";
import { contacts } from "./contacts";
import { leadSourceEnum } from "./contacts";
import { users } from "./users";

export const leadStatusEnum = pgEnum("lead_status", [
  "new",
  "contacted",
  "qualified",
  "unqualified",
  "converted",
]);

export const leads = pgTable(
  "leads",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    firstName: text("first_name").notNull(),
    lastName: text("last_name").notNull(),
    email: text("email").notNull().unique(),
    phone: text("phone"),
    companyName: text("company_name"),
    title: text("title"),
    source: leadSourceEnum("source"),
    status: leadStatusEnum("status").notNull().default("new"),
    convertedAt: timestamp("converted_at", { withTimezone: true }),
    convertedContactId: uuid("converted_contact_id").references(() => contacts.id, { onDelete: "set null" }),
    ownerId: uuid("owner_id").references(() => users.id, { onDelete: "set null" }),
    notes: text("notes"),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_leads_status").on(table.status),
    index("idx_leads_owner_id").on(table.ownerId),
    index("idx_leads_source").on(table.source),
  ]
);

File: schemas/crm/spacetimedb/leads.rs
// leads: Prospective contacts not yet converted into full contacts.
// See README.md for full design rationale.
// Uses LeadSource from contacts.rs — do not redefine here.

use spacetimedb::Timestamp;
use crate::contacts::LeadSource;

#[derive(SpacetimeType, Clone)]
pub enum LeadStatus {
    New, // type: String
    Contacted,
    Qualified,
    Unqualified,
    Converted,
}

#[spacetimedb::table(name = leads, public)]
pub struct Lead {
    #[primary_key]
    pub id: String, // UUID
    pub first_name: String,
    pub last_name: String,
    #[unique]
    pub email: String,
    pub phone: Option<String>,
    pub company_name: Option<String>,
    pub title: Option<String>,
    #[index(btree)]
    pub source: Option<LeadSource>,
    #[index(btree)]
    pub status: LeadStatus,
    pub converted_at: Option<Timestamp>,
    pub converted_contact_id: Option<String>, // UUID — FK → contacts.id (set null)
    #[index(btree)]
    pub owner_id: Option<String>, // UUID — FK → users.id (set null)
    pub notes: Option<String>,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
